src/components/location-aside.js

```diff
@@ -51,131 +53,378 @@ LayoutWithIcon.propTypes = {
   palette: PropTypes.string
 };
 
-export default function LocationAside ({ node }) {
-  const { field_phone_number: fieldPhoneNumber, field_email: fieldEmail, relationships } = node;
+export default function LocationAside ({ node, isStudySpaceAside = false }) {
+  const {
+    field_phone_number: fieldPhoneNumber,
+    field_room_number: fieldRoomNumber,
+    field_email: fieldEmail,
+    field_noise_level: noiseLevel,
+    field_space_features: spaceFeatures,
+    relationships } = node;
   const buildingNode = relationships?.field_room_building;
-  const parentLocationNode
-    = relationships?.field_parent_location?.relationships?.field_parent_location;
-  const locationNode = buildingNode ?? parentLocationNode ?? node;
-  const floorPlans = relationships?.field_floor_plan;
+  const parentLocationNode = relationships?.field_parent_location;
+  const locationNode = buildingNode ?? parentLocationNode?.relationships?.field_parent_location ?? node;
+  const locationTitle = buildingNode?.title ?? parentLocationNode?.title;
+  const floor = getFloor({ node });
+  const fid = node.relationships?.field_floor?.id;
+  const maybeFloorPlan = useFloorPlan(buildingNode?.id ?? parentLocationNode?.id, fid);
+  let floorPlans = relationships?.field_floor_plan;
+  if (floor && floorPlans.length === 0) {
+    floorPlans = maybeFloorPlan;
+  }
+  let normalizedFloorPlans = [];
+  if (Array.isArray(floorPlans)) {
+    normalizedFloorPlans = floorPlans;
+  } else if (floorPlans) {
+    normalizedFloorPlans = [floorPlans];
+  }
+
+  if (isStudySpaceAside) {
+    return (
+      <>
+        <StudySpaceLocationSection
+          locationTitle={locationTitle}
+          floor={floor}
+          roomNumber={fieldRoomNumber}
+          normalizedFloorPlans={normalizedFloorPlans}
+          maybeFloorPlan={maybeFloorPlan}
+        />
```

Comment:
```suggestion
        <StudySpaceLocationSection {...{
          floor,
          locationTitle,
          maybeFloorPlan,
          normalizedFloorPlans,
          roomNumber: fieldRoomNumber
        }}  />
```
When there are a lot of attributes, especially when the attribute name matches the value name, I sometimes like to set them using a spread operator. Not a blocker. Just showing a different approach. This is also untested.